From ab98ad397be43d16a0221f78408ed8dbc522e7af Mon Sep 17 00:00:00 2001
From: Benjamin Berg <benjamin.berg@intel.com>
Date: Mon, 25 Dec 2023 12:21:02 +0200
Subject: [PATCH] tests: Stop wpa_supplicant AP at end of VHT 80 MHz (US) test

The regulatory domain cannot be reset while the AP is still active. Add
the appropriate commands to bring down the AP after the test, which will
also make resetting the regulatory domain work.

Signed-off-by: Benjamin Berg <benjamin.berg@intel.com>
---
 tests/hwsim/test_wpas_ap.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tests/hwsim/test_wpas_ap.py b/tests/hwsim/test_wpas_ap.py
index 7090f23e4..67a34cd61 100644
--- a/tests/hwsim/test_wpas_ap.py
+++ b/tests/hwsim/test_wpas_ap.py
@@ -795,6 +795,8 @@ def run_wpas_ap_open_vht80_us(dev, freq, center_freq, ht40):
         if "WIDTH=80 MHz" not in sig:
             raise Exception("Unexpected SIGNAL_POLL value(2): " + str(sig))
     finally:
+        dev[0].request("REMOVE_NETWORK all")
+        dev[0].wait_disconnected()
         set_country("00")
         dev[0].set("country", "00")
         dev[1].flush_scan_cache()
-- 
2.43.0

